thrall/app/controllers/ReaperController.scala

```diff
@@ -0,0 +1,196 @@
+package controllers
+
+import akka.actor.Scheduler
+import com.gu.mediaservice.lib.ImageIngestOperations
+import com.gu.mediaservice.lib.auth.Permissions.DeleteImage
+import com.gu.mediaservice.lib.auth.{Authentication, Authorisation, BaseControllerWithLoginRedirects}
+import com.gu.mediaservice.lib.config.Services
+import com.gu.mediaservice.lib.elasticsearch.ReapableEligibility
+import com.gu.mediaservice.lib.logging.{GridLogging, MarkerMap}
+import com.gu.mediaservice.lib.metadata.SoftDeletedMetadataTable
+import com.gu.mediaservice.model.{ImageStatusRecord, SoftDeletedMetadata}
+import lib.{ThrallConfig, ThrallMetrics, ThrallStore}
+import lib.elasticsearch.ElasticSearch
+import org.joda.time.{DateTime, DateTimeZone}
+import play.api.libs.json.{JsValue, Json}
+import play.api.mvc.{Action, AnyContent, ControllerComponents}
+
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.{ExecutionContext, Future}
+import scala.jdk.CollectionConverters.collectionAsScalaIterableConverter
+import scala.language.postfixOps
+
+class ReaperController(
+  es: ElasticSearch,
+  store: ThrallStore,
+  authorisation: Authorisation,
+  config: ThrallConfig,
+  scheduler: Scheduler,
+  softDeletedMetadataTable: SoftDeletedMetadataTable,
+  metrics: ThrallMetrics,
+  override val auth: Authentication,
+  override val services: Services,
+  override val controllerComponents: ControllerComponents,
+)(implicit val ec: ExecutionContext) extends BaseControllerWithLoginRedirects with GridLogging {
+
+  private val CONTROL_FILE_NAME = "PAUSED"
+
+  private val INTERVAL = 15 minutes // based on max of 1000 per reap, this interval will max out at 96,000 images per day
+  private val REAPS_PER_WEEK = 7.days / INTERVAL
+
+  implicit val logMarker: MarkerMap = MarkerMap()
+
+  private def getIsReapable = new ReapableEligibility {
```

Comment:
I suppose it doesn't make a huge difference, but why `def` over `val` here?

Reply:
Because it need to be instantiated each time, since the 2 weeks old bit of it needs to be relative to the time it runs, each time

Reply:
Where is the 2 weeks bit? 

Also, can we make it more clear to future developers why this needs to be a `def`?